adapters/ownadx/ownadx.go

```diff
@@ -0,0 +1,165 @@
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return []*adapters.RequestData{{
+		Method:  http.MethodPost,
+		Body:    reqJSON,
+		Uri:     url,
+		Headers: headers,
+	}}
+}
+
+func (adapter *adapter) buildEndpointURL(params *openrtb_ext.ExtImpOwnAdx) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{
+		ZoneID:    params.SspId,
+		AccountID: params.SeatId,
+		SourceId:  params.TokenId,
+	}
+	return macros.ResolveMacros(adapter.endpoint, endpointParams)
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpOwnAdx, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "Bidder extension not valid or can't be unmarshalled",
+		}
+	}
+
+	var ownAdxExt openrtb_ext.ExtImpOwnAdx
+	if err := json.Unmarshal(bidderExt.Bidder, &ownAdxExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "Error while unmarshaling bidder extension",
+		}
+	}
+
+	return &ownAdxExt, nil
+}
+
+func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var ownAdxExt *openrtb_ext.ExtImpOwnAdx
+	ownAdxExt, err := getImpressionExt(&(request.Imp[0]))
+	if err != nil {
+		return nil, []error{
+			httpBadResponseError(fmt.Sprintf("Bidder extension not valid or can't be unmarshalled")),
+		}
+	}
+
+	endPoint, err := adapter.buildEndpointURL(ownAdxExt)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return getRequestData(reqJSON, endPoint), nil
+}
+func httpBadResponseError(message string) error {
+	return &errortypes.BadServerResponse{
+		Message: message,
+	}
+}
+
+func getBidType(ext bidExt) (openrtb_ext.BidType, error) {
+	return openrtb_ext.ParseBidType(ext.MediaType)
+}
+
+func (adapter *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{
+			httpBadResponseError(fmt.Sprintf("Bad request: %d", response.StatusCode)),
+		}
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{
+			httpBadResponseError(fmt.Sprintf("Unexpected http status code: %d", response.StatusCode)),
+		}
+	}
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{
+			httpBadResponseError(fmt.Sprintf("Bad server response ")),
+		}
+	}
+	if len(bidResp.SeatBid) == 0 {
+		return nil, []error{
+			httpBadResponseError(fmt.Sprintf("Array SeatBid cannot be empty ")),
```

Comment:
`httpBadResponseError()` only initalises `&errortypes.BadServerResponse`.  So just a suggestion to consider directly using `&errortypes.BadServerResponse` instead of `httpBadResponseError`.   

```
return nil, &errortypes.BadServerResponse{
		Message: fmt.Sprintf(....),
	}
}
```